visualization/app/codeCharta/core/data/data.service.ts

```diff
@@ -43,6 +43,28 @@ export class DataService {
 
     }
 
+    /**
+     *
+     * @param {CodeMapNode} map
+     * @param {Function} fn
+     * @param {any[]} parameters
+     *
+     * Function that goes through every node in a map and applies there a function
+     * whose parameters are hold in parameters array.
+     *
+     * The function input in fn is required to get their arguments into a single array, which allows to input
+     * as many variables as desired and also to remain the computed value in all calls to fn that recursive
+     * procedure requires.
+     */
+    public goThroughMap(map: CodeMapNode, fn: Function, parameters: any[]|{}){
```

Comment:
You can import "hierarchy" from d3 and use it to achieve the exact same functionality in a more readable way. It also makes this method unnecessary. Take a look at this [line](https://github.com/MaibornWolff/codecharta/blob/master/visualization/app/codeCharta/core/data/data.decorator.service.ts#L72)